MinitexPDFProtocols/MinitexPDFProtocols.swift

```diff
@@ -9,67 +9,118 @@
 import Foundation
 import UIKit
 
-// the items that are marked required are needed to
-// recreate an annotation in PSPDFKit later
-@objc public class MinitexPDFAnnotation: NSObject, Codable {
-  // required
-  public let pageIndex: UInt
-  // required
-  public let type: String
-  // required
-  public let bbox: String
-  // required
-  public let rects: [String]
 
-  // optional
-  public let color: String?
-  // optional
-  public let opacity: Float?
-  // optional
-  // we store this in case the field parsing fails, then we have the original JSON
-  public let JSONData: Data?
+/// A generic annotation in a PDF document.
+@objc public protocol MinitexPDFAnnotation: class {
+  /// An annotation must belong to one page of the PDF document.
```

Comment:
Based on Paul's note, this comment should be updated to denote either that the annotation is wholly contained to this page OR BEGINS on this page. Example, a highlight that may stretch between multiple pages.